### PR TITLE
CB-6803 Add Audit Service

### PR DESCRIPTION
**Pull request by bergerdenes**

### CODE REVIEW

File: templates/compose-audit.tmpl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            - DB_NAME={{{get . "AUDIT_DB_ENV_DB"}}}
+            - DB_USER=postgres
+            - DB_PASSWORD=""
+            - DB_DEBUG=true
+            - VAULT_ADDR=http://vault:{{{get . "VAULT_BIND_PORT"}}}
+            - KUBE_TOKEN_PATH=/etc/vault-root-token.txt
+            - SERVICEDEPENDENCIES_CLOUDBREAKENVHOST=cloudbreak-environment.cloudbreak
+            - SERVICEDEPENDENCIES_CLOUDBREAKENVPORT={{{get . "CB_PORT"}}}
+        volumes:
+            - ./etc/vault-dev-root-token.txt:/etc/vault-root-token.txt
```
Comment (keyki) [inline templates/compose-audit.tmpl:16]: How is this file getting created?

**Comment by keyki** [comment]
Not sure if we want to run this service from intellij at any point, in case we do we need to include it in the CB_LOCAL_DEV list. I don't see CB related changes, I'm assuming we don't have the integration yet so the ENV var is still unknown how we're going to refer this service from CB.

File: templates/compose-audit.tmpl (hunk 1 above)
Comment (bergerdenes) [reply templates/compose-audit.tmpl:16]: Nice catch. :D

**Comment by bergerdenes** [comment]
We do not want to run this service from intellij at all. This is an external one (like environment2) which happens to use the local DB, and that is the main point: to check the moving parts of auditing.
The launch of the "audit" service does not happen automatically, it is protected by a switch so for other developers it is simply non-existing or rather transparent.

File: include/vault.bash [hunk 2]
```diff
@@ -231,6 +233,12 @@ cloudbreak-delete-vault-data() {
...
     remove-variable-from-profile VAULT_ROOT_TOKEN
-    
+
     rm -rf $VAULT_CONFIG_FILE
 }
+
+create-token-file() {
+    mkdir -p ./etc
+    # token file is needed for Audit Service with a UUID inside it. This is temporary until Thunderhead codebase is updated to accept real root tokens.
+    uuidgen > "./etc/vault-dev-root-token.txt"
```
Comment (keyki) [inline include/vault.bash:243]: How is this going to work with Vault?

Comment (keyki) [reply include/vault.bash:243]: NVM. I see your comment.

Comment (bergerdenes) [reply include/vault.bash:243]: It won't work with our Vault as of now. This file is just needed for Audit Service to boot up.

**Pull request merged**